### PR TITLE
Describe how to set up Athena with S3 data

### PR DESCRIPTION
**Pull request by thijs-obj**

### CODE REVIEW

File: docs/docs/tracking/collector/s3.md [hunk 1]
```diff
@@ -97,3 +95,217 @@ raw topic, to be picked up by Snowplow's enrichment.
...
+All Athena query results are written to S3, to the so called 'S3 staging'. Therefore, an s3 bucket or prefix
+should be created that can be used for that purpose. This S3 prefix should not overlap with the location of
+the data.
+
+An account should be setup, with the following access:
+* S3:
+  * read access to the data, i.e. the `LOCATION` that the `events` table gets its data from.
+  * write access to the S3 staging prefix.
+* Athena: access to query/create/etc. tables.
+* Glue: access to Glue data catalog.
```
Comment (thijs-obj) [inline docs/docs/tracking/collector/s3.md:269]: I want to do this a bit more in detail, similar to what was done here: https://github.com/objectiv/objectiv-analytics/blob/main/bach/tests/athena-setup-test-db.md. But that requires some more research. I think the information we have here now is sufficient for now.

File: docs/docs/tracking/collector/s3.md [hunk 2]
```diff
@@ -97,3 +95,217 @@ raw topic, to be picked up by Snowplow's enrichment.
 
 To check if the messages have successfully arrived in the queue, please review to monitoring in the AWS 
 console. Events should show up as either `PutRecords` (Kinesis) or `Number of messages received (SQS).
+
```
Comment (thijs-obj) [inline docs/docs/tracking/collector/s3.md:98]: I have the feeling that there are some steps missing above. It seems the steps above get the data in SQS, but not quite to S3. That's @borft 's expertise tho.

File: docs/docs/tracking/collector/s3.md [hunk 3]
```diff
@@ -97,3 +95,217 @@ raw topic, to be picked up by Snowplow's enrichment.
...
+Additionally, there are optional query string parameters that can be added to the url:
+* `work_group`: work-group under which the queries run
+* `catalog_name`: catalog that contains the table defintions
+
+Note that all values must be properly escaped. As a help, the modelhub offers a function that constructs the
+url: `athena_construct_engine_url()`.
+
+```python
+from modelhub import athena_construct_engine_url
```
Comment (thijs-obj) [inline docs/docs/tracking/collector/s3.md:295]: This function will soon be added: https://github.com/objectiv/objectiv-analytics/pull/1416

File: docs/docs/tracking/collector/s3.md (hunk 2 above)
Comment (borft) [reply docs/docs/tracking/collector/s3.md:98]: The steps that are missing involve setting up a working Snowplow pipeline. I'd say, at least at the time, that was considered beyond the scope of this document.

FYI: what's happening is:
objectiv-collector -> MQ -> enrich-kinesis -> MQ -> s3-loader -> s3

The part beyond the Objectiv Collector is considered to be part of the Snowplow pipeline

File: docs/docs/tracking/collector/s3.md [hunk 4]
```diff
@@ -97,3 +95,217 @@ raw topic, to be picked up by Snowplow's enrichment.
 
 To check if the messages have successfully arrived in the queue, please review to monitoring in the AWS 
 console. Events should show up as either `PutRecords` (Kinesis) or `Number of messages received (SQS).
+
+
+## Accessing data with Athena
```
Comment (ivarpruijn) [inline docs/docs/tracking/collector/s3.md:100]:
```suggestion
## Use Objectiv with Athena
```

File: docs/docs/tracking/collector/s3.md [hunk 5]
```diff
@@ -97,3 +95,217 @@ raw topic, to be picked up by Snowplow's enrichment.
...
+To be able to use the data on S3 with Bach and modelhub, a few steps need to be taken:
```
Comment (ivarpruijn) [inline docs/docs/tracking/collector/s3.md:101]:
```suggestion
To be able to use the data on S3 with [Bach](/modeling/bach/index.mdx) and [the open model hub](/modeling/open-model-hub/index.mdx), a few steps need to be taken:
```

File: docs/docs/tracking/collector/s3.md [hunk 6]
```diff
@@ -97,3 +95,217 @@ raw topic, to be picked up by Snowplow's enrichment.
...
+ 1. An Athena table must be created that makes the S3 data queryable.
+ 2. Additional Athena setup: permissions and query-result storage
+ 3. Constructing the URL to connect with Athena.
```
Comment (ivarpruijn) [inline docs/docs/tracking/collector/s3.md:104]:
```suggestion
 1. Create an Athena table that makes the S3 data queryable.
 2. Perform additional Athena setup: permissions and query-result storage.
 3. Construct the URL to connect with Athena.
```

**Review by ivarpruijn — APPROVED** [review]
A few small suggestions, but LGTM

**Comment by ivarpruijn** [comment]
@thijs-obj let's make this PR merge into the release PR I'm working on for Athena support: https://github.com/objectiv/objectiv.io/pull/594

File: docs/docs/tracking/collector/s3.md (hunk 2 above)
Comment (thijs-obj) [reply docs/docs/tracking/collector/s3.md:104]: After our chat I added a little bit more info in 5c3e3de834cb1b5e3fc3e961c970607180cd7f8c

File: docs/docs/tracking/collector/s3.md [hunk 7]
```diff
@@ -95,5 +99,222 @@ This indicates that the Collector will try to push events. If this fails, loggin
...
+* `region_name`: region, e.g. `eu-west-1`
+* `schema_name`: name of database that contains the `events` table defined earlier
+* `s3_staging_dir`: path to S3 where query results can be stored
+
+Additionally, there are optional query string parameters that can be added to the url:
+* `work_group`: work-group under which the queries run
+* `catalog_name`: catalog that contains the table definitions
+
+If the access key fields are not set, then the Athena library will use default mechanisms to get
+credentials (environment variables, config files, etc. See [Amazon docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials))
```
Comment (thijs-obj) [inline docs/docs/tracking/collector/s3.md:298]: Added here some info on not providing the credentials @borft

**Comment by thijs-obj** [comment]
> @thijs-obj let's make this PR merge into the release PR I'm working on for Athena support: https://github.com/objectiv/objectiv.io/pull/594

Done

**Comment by thijs-obj** [comment]
Made some small changes after the first review round. If everybody is happy please go ahead and merge this. Otherwise I'll work more on this on Monday

**Comment by ivarpruijn** [comment]
> Made some small changes after the first review round. If everybody is happy please go ahead and merge this. Otherwise I'll work more on this on Monday

All LGTM

**Comment by thijs-obj** [comment]
No formal approval by all reviewers. But I think everyone is OK? Let me know otherwise and I'll make additional changes. Will merge for now

**Pull request merged**